c/meterpreter/source/extensions/priv/namedpipe_efs.c

```diff
@@ -50,6 +50,22 @@ DWORD elevate_via_namedpipe_efs(Remote* remote, Packet* packet)
 			BREAK_ON_ERROR("[ELEVATE] elevate_via_namedpipe_efs: Windows versions older than 6.0 are unsupported");
 		}
 
+		// Windows Vista / Server 2008 (6.0 and 6.1) only supports \pipe\lsarpc endpoint
+		if (os.dwMajorVersion == 6 && (os.dwMajorVersion == 0 || os.dwMajorVersion == 1)) {
```

Comment:
```suggestion
		if (os.dwMajorVersion < 6) || ((os.dwMajorVersion == 6 && (os.dwMajorVersion == 0 || os.dwMajorVersion == 1))) {
```

Update since this will also make sure this applies to older versions as well, otherwise you will be trying do do `efsrpc` on systems that never were designed to have that pipe in the first place.

Reply:
The technique (PetitPotam EfsRpc) only supports **2008 and upper**. I'm not sure about behavior on downer versions.

Reply:
applied this on [5d8a162](https://github.com/rapid7/metasploit-payloads/pull/625/commits/5d8a162cf96e0f6611a6a66d97deb56f7e3f2c9a)